envpool/sokoban/sokoban_node.cc

```diff
@@ -157,12 +157,26 @@ float SokobanNode::GoalDistanceEstimate(SokobanNode& goal_node) {
   float h = 0;
   for (const auto& box : boxes) {
     float min_distance = std::numeric_limits<float>::max();
+    bool goal_along_x = false;
+    bool goal_along_y = false;
     for (const auto& goal_box : goal_node.boxes) {
       float distance =
           abs(box.first - goal_box.first) + abs(box.second - goal_box.second);
       min_distance = std::min(min_distance, distance);
+      if (box.first == goal_box.first) {
+        goal_along_y = true;
+      }
+      if (box.second == goal_box.second) {
+        goal_along_x = true;
+      }
     }
     h += min_distance;
+    int surr_walls = SurroundingWalls(box);
```

Comment:
I don't understand this heuristic. If there's 1 wall next to the box it can still be moved just fine. E.g.:


```
...
.@#
...
```

first we'd move the box down and then we can move it freely.